interface/ceed.c

```diff
@@ -910,6 +920,43 @@ int CeedRestoreJitSourceRoots(Ceed ceed, const char ***jit_source_roots) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Retrieve list of additional JiT defines from `Ceed` context.
+
+  Note: The caller is responsible for restoring `jit_defines` with @ref CeedRestoreJitDefines().
+
+  @param[in]  ceed            `Ceed` context
+  @param[out] num_jit_defines Number of JiT defines
+  @param[out] jit_defines     Strings such as `foo=bar`, used as `-Dfoo=bar` in JiT
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedGetJitDefines(Ceed ceed, CeedInt *num_defines, const char ***jit_defines) {
+  Ceed ceed_parent;
+
+  CeedCall(CeedGetParent(ceed, &ceed_parent));
+  *num_defines = ceed_parent->num_jit_defines;
+  *jit_defines = (const char **)ceed_parent->jit_defines;
+  return CEED_ERROR_SUCCESS;
+}
+
+/**
+  @brief Restore list of additional JiT defines from with @ref CeedGetJitDefines()
+
+  @param[in]  ceed        `Ceed` context
+  @param[out] jit_defines String such as `foo=bar`, used as `-Dfoo=bar` in JiT
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedRestoreJitDefines(Ceed ceed, const char ***jit_defines) {
+  *jit_defines = NULL;
+  return CEED_ERROR_SUCCESS;
+}
```

Comment:
Not sure how I feel about a Restore function that doesn't actually do anything? Is there something we would plan on doing later with this function?

Reply:
My long-term plan is to make any get that isn't a scalar value need a restore so we don't have to remember the magic list of exceptions in our head

Reply:
Gotcha. Should we add a lock or something to kinda enforce it? We don't have to remember the exception, but there's no consequence for not remembering to use it at all.

Reply:
Maybe a reader count? It could be bad to miss an option added last minute 

I should put in a branch to tidy all this up with all our Get*

Reply:
Yeah, reader count sounds good.

Reply:
> It could be bad to miss an option added last minute

You mean don't allow `CeedAddJitDefine` to be called if the reader count is non-zero?

Reply:
See c6d6a888b8f8436f07194058d4fda63c5f1fa761

Reply:
Looks good to me. Small suggestion for the error message.